src/gui/guides.c

```diff
@@ -475,25 +571,275 @@ GList *dt_guides_init()
+  gtk_box_pack_start(GTK_BOX(vbox), GTK_WIDGET(gw->colors), TRUE, TRUE, 0);
+
+  gtk_container_add(GTK_CONTAINER(pop), vbox);
+  gtk_widget_show_all(pop);
+}
+
+void dt_guides_update_button_state()
+{
+  if(!darktable.view_manager) return;
+  GtkWidget *bt = darktable.view_manager->guides_toggle;
+
+  gchar *key = _conf_get_path("global", "show", NULL);
+  gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(bt), dt_conf_get_bool(key));
+  g_free(key);
+}
+
+void dt_guides_button_toggled()
+{
+  gchar *key = _conf_get_path("global", "show", NULL);
+  dt_conf_set_bool(key, !dt_conf_get_bool(key));
+  g_free(key);
+}
+
+void dt_guides_draw(cairo_t *cr, const float left, const float top, const float width, const float height,
+                    const float zoom_scale)
+{
+  const double dashes = DT_PIXEL_APPLY_DPI(5.0) / zoom_scale;
+
+  dt_iop_module_t *module = darktable.develop->gui_module;
+
+  // first, we check if we need to show the guides or not
+  gchar *key = _conf_get_path("global", "show", NULL);
+  gboolean show = dt_conf_get_bool(key);
+  g_free(key);
+
+  // otherwise we try the module autoshow
+  if(!show && module)
+  {
+    key = _conf_get_path(module->op, "autoshow", NULL);
+    show = dt_conf_get_bool(key);
+    g_free(key);
+  }
+
+  // if it's still false, then nothing to draw
+  if(!show) return;
+
+  // let's get the guide to show
+  dt_guides_t *guide = _conf_get_guide("global");
+  if(!guide) return;
+
+  int flip = 0;
+  // retrieve guide flip
+  if(guide->support_flip)
+  {
+    key = _conf_get_path("global", guide->name, "flip");
+    if(dt_conf_key_exists(key)) flip = dt_conf_get_int(key);
+    g_free(key);
+  }
+
+  // save context
+  cairo_save(cr);
+  cairo_rectangle(cr, left, top, width, height);
+  cairo_clip(cr);
+  cairo_set_line_width(cr, DT_PIXEL_APPLY_DPI(1.0) / zoom_scale);
+  dt_draw_set_color_overlay(cr, 0.8, 1.0);
+  cairo_set_dash(cr, &dashes, 1, 0);
+
+  // Move coordinates to local center selection.
+  cairo_translate(cr, (width / 2 + left), (height / 2 + top));
+
+  // Flip horizontal.
+  if(flip == 1 || flip == 3) cairo_scale(cr, -1, 1);
+  // Flip vertical.
+  if(flip == 2 || flip == 3) cairo_scale(cr, 1, -1);
+
+  // we do the drawing itself
+  guide->draw(cr, -width / 2.0, -height / 2.0, width, height, zoom_scale, guide->user_data);
+
+  cairo_stroke_preserve(cr);
+  cairo_set_dash(cr, &dashes, 0, 0);
+  dt_draw_set_color_overlay(cr, 0.3, 0.8);
+  cairo_stroke(cr);
+
+  cairo_restore(cr);
+}
+
+static void _settings_autoshow_change(GtkWidget *mi, dt_iop_module_t *module)
+{
+  // we inverse the autoshow value for the module
+  gchar *key = _conf_get_path(module->op, "autoshow", NULL);
+  dt_conf_set_bool(key, !dt_conf_get_bool(key));
+  g_free(key);
+  dt_control_queue_redraw_center();
+}
+
+void dt_guides_add_module_menuitem(void *menu, struct dt_iop_module_t *module)
+{
+  GtkWidget *mi = gtk_check_menu_item_new_with_label(_("show masks"));
```

Comment:
Should be "show guides". Masks are something else. The checkbox needs some padding around it too.

Reply:
good catch !
the most worrying thing is that I was sober :)

I've added some css margins too, but better review that. You're better than me to set right values ;)

Reply:
Yeah that looks better thanks. And as a bonus I found another unrelated issue while I was testing (I tagged you in it).